### PR TITLE
ios_acls - unable to render range commands under protocol_options and fragment type changed

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

1. Unable to render range attribute under protocol_option attribute.
2. Deprecated fragment key of type str and added  enable_fragment of type bool.
3. Reserved keyword used as a variable - corrected
4. Updated Parsed state documentation
5. Tests updated

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ios_acls

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by KB-perByte** [comment]
recheck

File: docs/cisco.ios.ios_acls_module.rst [hunk 1]
```diff
@@ -3857,7 +3880,7 @@ Parameters
...
-                        <div>The state <em>parsed</em> reads the configuration from <code>running_config</code> option and transforms it into JSON format as per the resource module parameters and the value is returned in the <em>parsed</em> key within the result. The value of <code>running_config</code> option should be the same format as the output of command <em>show access-list</em> executed on device. For state <em>parsed</em> active connection to remote host is not required.</div>
+                        <div>The state <em>parsed</em> reads the configuration from <code>running_config</code> option and transforms it into JSON format as per the resource module parameters and the value is returned in the <em>parsed</em> key within the result. The value of <code>running_config</code> option should be the same format as the output of commands <em>show access-list</em> and <em>show running-config | include ip(v6</em>* access-list|remark) executed on device. For state <em>parsed</em> active connection to remote host is not required.</div>
```
Comment (NilashishC) [inline docs/cisco.ios.ios_acls_module.rst:3883]: Can you also add how the two configs should be arranged?

File: plugins/module_utils/network/ios/rm_templates/acls.py [hunk 2]
```diff
@@ -335,11 +349,15 @@ def __init__(self, lines=None):
...
+                                            dest_port_protocol.split(' ')[1] if dest_port_protocol is defined else None }}",
+                                        "range": {
+                                            "start": "{{ drange_start if drange_start is defined else None }}",
+                                            "end": "{{ drange_end if drange_end is defined else None }}",
+                                        },
                                     },
                                 },
                                 "dscp": "{{ dscp }}",
-                                "fragments": "{{ fragments }}",
+                                "enable_fragments": "{{ True if enable_fragments is defined }}",
```
Comment (NilashishC) [inline plugins/module_utils/network/ios/rm_templates/acls.py:360]: Missing else clause.

File: plugins/module_utils/network/ios/rm_templates/acls.py [hunk 3]
```diff
@@ -335,11 +349,15 @@ def __init__(self, lines=None):
...
                                 "log": {
                                     "set": "{{ True if log is defined and 'tag' not in log }}",
```
Comment (NilashishC) [inline plugins/module_utils/network/ios/rm_templates/acls.py:362]: Missing else clause.

**Review by NilashishC — APPROVED** [review]

File: tests/unit/modules/network/ios/test_ios_acls.py [hunk 4]
```diff
@@ -743,7 +743,9 @@ def test_ios_acls_parsed(self):
                             {
                                 "destination": {
                                     "any": True,
-                                    "port_protocol": {"eq": "telnet"},
+                                    "port_protocol": {
+                                        "range": {"start": 10, "end": 20}
```
Comment (ashwini-mhatre) [inline tests/unit/modules/network/ios/test_ios_acls.py:747]: instead of removing this can you add one more separate test for range.

**Review by ashwini-mhatre — APPROVED** [review]

**Pull request merged**